gate-oauth2/src/main/groovy/com/netflix/spinnaker/gate/security/oauth2/SpinnakerUserInfoTokenServices.groovy

```diff
@@ -237,6 +238,11 @@ class SpinnakerUserInfoTokenServices implements ResourceServerTokenServices {
     }
     def roles = details[userInfoMapping.roles] ?: []
     if (roles instanceof Collection) {
+      // Some providers (Azure AD) return roles in this format: ["[\"role-1\", \"role-2\"]"]
+      if (roles[0][0] == '[' || roles[0][0] == '{' ) {
```

Comment:
why `{`? If this was a JSON object what would the expectation be after parsing and flattening?

Reply:
I think it's incredibly unlikely that there would be a `{`, but then again, I figured we're already handling bad behavior, so maybe try to handle all bad behavior?

Good point, though.  I'll remove the `{`.